src/DataProvider/DefaultDataProvider.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMeilisearchPlugin\DataProvider;
+
+use Doctrine\Persistence\ManagerRegistry;
+use DoctrineBatchUtils\BatchProcessing\SelectBatchIteratorAggregate;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Setono\Doctrine\ORMTrait;
+use Setono\SyliusMeilisearchPlugin\Config\Index;
+use Setono\SyliusMeilisearchPlugin\Event\DataProviderEvent;
+use Setono\SyliusMeilisearchPlugin\Model\IndexableInterface;
+
+final class DefaultDataProvider implements DataProviderInterface
+{
+    use ORMTrait;
+
+    public function __construct(
+        ManagerRegistry $managerRegistry,
+        private readonly EventDispatcherInterface $eventDispatcher,
+    ) {
+        $this->managerRegistry = $managerRegistry;
+    }
+
+    public function getObjects(string $entity, Index $index): \Generator
+    {
+        $qb = $this
+            ->getManager($entity)
+            ->createQueryBuilder()
+            ->select('o')
+            ->from($entity, 'o');
+
+        $this->eventDispatcher->dispatch(new DataProviderEvent($entity, $index, $qb));
```

Comment:
I have mixed feelings about the event's name. The event name should reflect a something that happened in the past. So maybe something like `DataObjectProvided` or `PreDataObjectProvided` if you want allow to e.g. modify a $qb?

Reply:
Yeah, agreed :)